src/PostgREST/Middleware.hs

```diff
@@ -172,3 +174,8 @@ optionalRollback AppConfig{..} ApiRequest{..} transaction = do
             [(HTTP.hPreferenceApplied, BS.pack (show Rollback))]
       | otherwise =
           identity
+
+-- | Do a pg set_config(setting, value, true) call. This is equivalent to a SET LOCAL.
+setConfigLocal :: Text -> (Text, Text) -> H.Snippet
+setConfigLocal prefix (k, v) =
+  "set_config(" <> unknownLiteral (prefix <> k) <> ", " <> unknownLiteral v <> ", true)"
```

Comment:
I don't think this belongs here. I guess you moved it here, because of the `runPgLocals` function above. Can we move that into a `where` in that function to make it clear where it belongs?

We will end up splitting up Middleware anyway - it's one of those generics...

Reply:
Done!